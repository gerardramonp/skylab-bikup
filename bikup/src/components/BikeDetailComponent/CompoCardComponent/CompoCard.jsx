import React from 'react';
import './CompoCard.scss';
import { NavLink } from 'react-router-dom';

function CompoCard({ compoInfo, bikeName }) {
    console.log(compoInfo);
    return (
        <div className="compocard">
            <div className="compocard__content">
                <img
                    src="https://image.flaticon.com/icons/svg/2623/2623442.svg"
                    alt="compo img"
                    className="compocard__img"
                />
                <div className="compocard__info">
                    <div className="compocard__labels">
                        <p>{compoInfo.compoDisplayName}</p>

                        <p className="labels__status">
                            ({compoInfo.compoAccumulatedMeters / 1000} /
                            {compoInfo.compoLife / 1000} Km)
                        </p>
                    </div>
                    <div>
                        <progress
                            id="progress"
                            value={compoInfo.compoAccumulatedMeters / 1000}
                            max={compoInfo.compoLife / 1000}
                        ></progress>
                    </div>
                </div>

                <NavLink
                    className="compocard__button"
                    to={`/bikes/${bikeName.replace(
                        / /g,
                        ''
                    )}/${compoInfo.compoDisplayName.replace(/ /g, '')}`}
                >
                    <img
                        className="button__img"
                        src="https://image.flaticon.com/icons/svg/271/271228.svg"
                        alt=">"
                    />
                </NavLink>
            </div>
        </div>
    );
}

export default CompoCard;
