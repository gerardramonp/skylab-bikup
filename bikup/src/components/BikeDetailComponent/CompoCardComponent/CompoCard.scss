$light-bg-color: #eeeeee;
$light-highlight-color: #d7d7d7;
$main-color: #32e0c4;

.compocard {
    display: flex;
    flex-direction: row;
    margin-bottom: 18px;
    width: 88%;
    background-color: $light-highlight-color;
    border-radius: 5px;
}

.card-decoration-1 {
    height: inherit;
    width: 2%;
    border-top-left-radius: 5px;
    border-bottom-left-radius: 5px;
    background-color: $main-color;
}

.compocard__img {
    height: 30px;
    width: 30px;
    margin-right: 12px;
}

.compocard__content {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    width: 98%;
    padding: 6px 12px;
}

.compocard__info {
    display: flex;
    flex-direction: column;
}

.compocard__labels {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
}

.labels__status {
    font-size: 10px;
}

.compocard__button {
    display: flex;
    align-items: center;
    background-color: $main-color;
    border: none;
    padding: 12px;
    border-radius: 5px;
}

.button__img {
    height: 16px;
}
