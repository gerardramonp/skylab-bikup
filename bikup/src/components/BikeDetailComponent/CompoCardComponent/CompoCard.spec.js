import React from 'react';
import renderer from 'react-test-renderer';
import CompoCard from './CompoCard';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Compo Card', () => {
	let tree = null;

	let compoInfo = {
		compoDisplayName: '',
		compoAccumulatedMeters: 10,
		compoLife: 1,
	};
	const bikeName = 'alma';

	beforeEach(() => {
		tree = renderer.create(
			<Router>
				<CompoCard compoInfo={compoInfo} bikeName={bikeName} />
			</Router>
		);
	});

	it('should match', () => {
		expect(tree.toJSON()).toMatchSnapshot();
	});

	it('Should execute function callback when clicking', () => {
		console.log(tree.find('button'));
	});
});
