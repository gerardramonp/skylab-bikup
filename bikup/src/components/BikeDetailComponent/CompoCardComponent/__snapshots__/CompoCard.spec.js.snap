// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compo Card should match 1`] = `
<div
  className="compocard"
>
  <div
    className="compocard__content"
  >
    <img
      alt="compo img"
      className="compocard__img"
      src="https://image.flaticon.com/icons/svg/2623/2623442.svg"
    />
    <div
      className="compocard__info"
    >
      <div
        className="compocard__labels"
      >
        <p>
          
        </p>
        <p
          className="labels__status"
        >
          (
          0.01
           /
          0.001
           Km)
        </p>
      </div>
      <div>
        <progress
          id="progress"
          max={0.001}
          value={0.01}
        />
      </div>
    </div>
    <a
      aria-current={null}
      className="compocard__button"
      href="/bikes/alma/"
      onClick={[Function]}
    >
      <img
        alt=">"
        className="button__img"
        src="https://image.flaticon.com/icons/svg/271/271228.svg"
      />
    </a>
  </div>
</div>
`;
